src/system/process.py

```diff
@@ -70,19 +71,19 @@ def terminate(self):
         return self.return_code
 
     @property
-    def started(self):
+    def started(self) -> bool:
         return True if self.process else False
 
     @property
-    def pid(self):
+    def pid(self) -> int:
         return self.process.pid if self.started else None
 
     @property
-    def stdout_data(self):
+    def stdout_data(self) -> Union[Text, bytes]:
```

Comment:
Since `Process` is our construct, we should return a consistent types by converting when needed.  This makes it easier for our usage of this class to be consistent.  I think `stdout_data(self) -> str:` is the easiest usage for our codebase.

Note; I think we should redesign the all the return types in this class to be based on our usage requirements.  Otherwise there are runtime behaviors become unpredictable.  If the intent is to keep the class type scheme as is we need a bunch of additional unit tests to cover these scenarios

Reply:
I have removed Union and updated the function to return `Any` here. `str` won't work because of how the function behaves. 

Error that pops up when returning `str` 
```bash
src/system/process.py:80: error: Returning Any from function declared to return "str"  [no-any-return]
```

Reply:
This error is because on line 20-21 these types are declared as any, they should be a specific type

Reply:
I cannot have them as `str`. This is because `this.stdout` is of type `_TemporaryFileWrapper`(line 26 and 27). It is later used to call the function `read()` and `close()` (line 80). `_TemporaryFileWrapper` is a private class. So I marked `this.stdout` as Any.

Reply:
By changing how the function works to prevent a leaky abstraction would work around this problem.

Note; While I think the design of this function could use some improvement, I also am not aware of problems related to how it works that should prevent this CR from merging, might be worthwhile to create an issue to resolve if/when it becomes more of a problem.